File: examples/cs/producer_consumer/Consumer.cs
using System;
using Dragonfly;


namespace Consumer
{
    class Program
    {
        static void Main(string[] args)
        {
            Module mod = new Module();
            mod.ConnectToMMM(MID.CONSUMER, "localhost:7111");
            mod.Subscribe(MT.TEST_DATA);

            Console.WriteLine("Consumer running...");

            Message m;
            while ( true)
            {
                mod.SendSignal(MT.REQUEST_TEST_DATA);
                Console.WriteLine("\nSent request for data");

                m = mod.ReadMessage(Module.ReadType.Blocking);

                if (m.msg_type == MT.TEST_DATA)
                {
                    Console.WriteLine("Received message {0}", m.msg_type);

                    object o;
                    MDF.TEST_DATA data = new MDF.TEST_DATA();
                    o = data;
                    m.GetData(ref o);

                    Console.WriteLine("Data   a: {0},  b: {1},  x: {2} ", data.a, data.b, data.x);
                }
            }

            mod.DisconnectFromMMM();
        }
    }
}

File: examples/matlab/producer_consumer/Consumer.m
function Consumer(mm_ip)

    global DF;

    Dragonfly_BaseDir = '../../..';
    addpath([Dragonfly_BaseDir '/lang/matlab']);

    MessageTypes =  { 'TEST_DATA' };

    ConnectArgs = {'CONSUMER', Dragonfly_BaseDir, ['./message_defs.mat']};
    if exist('mm_ip','var') && ~isempty(mm_ip)
        ConnectArgs{end+1} = ['-server_name ' mm_ip];
    end

    ConnectToMMM(ConnectArgs{:});
    Subscribe( MessageTypes{:})

    disp 'Consumer running..'

    while( 1)
        M = ReadMessage( 'blocking');
        switch(M.msg_type)
            case 'TEST_DATA'
                fprintf('Received message %s  Data = [a:%d  b:%d  x:%f]\n', M.msg_type, M.data.a, M.data.b, M.data.x);
        end
    end

    DisconnectFromMMM

